src/test/java/org/folio/calendar/integration/BaseApiTest.java

```diff
@@ -0,0 +1,210 @@
+@ContextConfiguration(initializers = { WireMockInitializer.class })
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public abstract class BaseApiTest {
+
+  public static final String TENANT_ID = "test";
+
+  @Getter
+  @Setter
+  protected static boolean initialized = false;
+
+  @Autowired
+  protected WireMockServer wireMockServer;
+
+  @Autowired
+  protected FolioModuleMetadata metadata;
+
+  @Autowired
+  protected TranslationService translationService;
+
+  @Value("${x-okapi-url}")
+  protected String okapiUrl = null;
+
+  @LocalServerPort
+  protected Integer port;
+
+  protected final OpenApiValidationFilter validationFilter = new OpenApiValidationFilter(
+    "api/calendar.yaml"
+  );
+
+  @BeforeEach
+  void clearCurrentDateOverride() {
+    DateUtils.setCurrentDateOverride(null);
+  }
+
+  @BeforeEach
+  void proxyLogTestStart(TestInfo testInfo) {
+    if (System.getenv().getOrDefault("PROXY_ENABLE", "false").equals("true")) {
+      String path = "/_/tests/";
+      if (testInfo.getTestClass().isPresent()) {
+        path += testInfo.getTestClass().get().getSimpleName() + "/";
+      } else {
+        path += "unknown/";
+      }
+      if (testInfo.getTestMethod().isPresent()) {
+        path += testInfo.getTestMethod().get().getName();
+      } else {
+        path += "unknown";
+      }
+      ra(ValidationSchema.NONE).get(getRequestUrl(path));
+    }
+  }
+
+  @AfterEach
+  void proxyLogTestFinish() {
+    if (System.getenv().getOrDefault("PROXY_ENABLE", "false").equals("true")) {
+      ra(ValidationSchema.NONE).get(getRequestUrl("/_/tests/_/finish"));
+    }
+  }
+
+  @BeforeAll
+  static void initialize() {
+    Locale.setDefault(Locale.US);
+
+    log.info("Configuring JSON to parse decimals as doubles, not floats");
+    // allow comparison with doubles, not floats
+    JsonConfig jsonConfig = JsonConfig
+      .jsonConfig()
+      .numberReturnType(JsonPathConfig.NumberReturnType.DOUBLE);
+    RestAssured.config = RestAssured.config().jsonConfig(jsonConfig);
+
+    log.info("Configuring JSON date mapping");
+    RestAssured.config =
+      RestAssured
+        .config()
+        .objectMapperConfig(
+          ObjectMapperConfig
+            .objectMapperConfig()
+            .jackson2ObjectMapperFactory((a, b) -> MapperUtils.MAPPER)
+        );
+
+    if (System.getenv().getOrDefault("PROXY_ENABLE", "false").equals("true")) {
+      String host = System.getenv().getOrDefault("PROXY_HOST", "localhost");
+      int port = Integer.parseInt(System.getenv().getOrDefault("PROXY_PORT", "8888"));
+      String scheme = System.getenv().getOrDefault("PROXY_SCHEME", "http");
+
+      log.info(String.format("Configuring proxy to %s://%s:%d", scheme, host, port));
+
+      RestAssured.proxy = new ProxySpecification(host, port, scheme);
+    }
+  }
+
+  @BeforeEach
+  public void createDatabase() {
+    if (!isInitialized()) {
+      tenantInstall(new TenantAttributes().moduleTo("mod-calendar"));
```

Comment:
What creates the database needed prior to the request being made to the Tenant API?

Reply:
The `@AutoConfigureEmbeddedDatabase` annotation does some spring magic with Docker

Reply:
Ah, thanks for explaining that.

You may have recognised a pattern in some of my feedback / questions. I tend to be somewhat cautious over magic (and especially for things like private field level auto wiring)

Reply:
How is the name of the database configured?

Are those values injected back into the app as environment variables / spring config?

Reply:
Spring really loves its magic autowiring (it took me quite a while to get used to it, myself).

The name of the database is generated by the `TENANT_ID` at the top of the `BaseApiTest`, plus some logic in `folio-spring-base` (to append the module name), making `test_mod_calendar`.

FSB handles all the database connectivity itself, and the embedded database starts a postgres server in docker during the application start, providing host/port/etc.  I'm not sure too much how it gets injected, I think the `@AutoConfigureEmbeddedDatabase` handles all of that.